article/app/controllers/ArticleController.scala

```diff
@@ -104,26 +110,35 @@ object ArticleController extends Controller with RendersItemResponse with Loggin
       renderFormat(htmlResponse, jsonResponse, article, Switches.all)
   }
 
-  def renderArticle(path: String, lastUpdate: Option[String], rendered: Option[Boolean]) = {
+  def renderArticle(path: String, lastUpdate: Option[String], rendered: Option[Boolean], page: Option[String]) = {
     if (LongCacheSwitch.isSwitchedOn) Action.async { implicit request =>
       // we cannot sensibly decache memcached (does not support surogate keys)
       // so if we are doing the 'soft purge' don't memcache
-      loadArticle(path, lastUpdate, rendered)
+      loadArticle(path, lastUpdate, rendered, page)
     } else MemcachedAction { implicit request =>
-      loadArticle(path, lastUpdate, rendered)
+      loadArticle(path, lastUpdate, rendered, page)
     }
   }
 
-  private def loadArticle(path: String, lastUpdate: Option[String], rendered: Option[Boolean])(implicit request: RequestHeader): Future[Result] = {
+  private def loadArticle(path: String, lastUpdate: Option[String], rendered: Option[Boolean], page: Option[String])(implicit request: RequestHeader): Future[Result] = {
     mapModel(path) { model =>
-      (lastUpdate, rendered) match {
-        case (Some(lastUpdate), _) => renderLatestFrom(model, lastUpdate)
-        case (None, Some(false)) => blockText(model, 6)
-        case (_, _) => render(path, model)
+      (lastUpdate, rendered, page) match {
+        case (Some(lastUpdate), _, _) => renderLatestFrom(model, lastUpdate)
+        case (None, Some(false), _) => blockText(model, 6)
+        case (_, _, Some(Int(pageNo))) => render(path, model, Some(pageNo))
+        case (_, _, _) => render(path, model, None)
       }
     }
   }
 
+  object Int {
+    def unapply(s : String) : Option[Int] = try {
```

Comment:
I'm not a big fan of custom `unapply`s. I'm not convinced their usefulness outweighs the amount of explanation required to those who haven't seen them before.